numpy/lib/tests/test_function_base.py

```diff
@@ -825,10 +824,10 @@ def test_slices(self):
     def test_fancy(self):
         # Deprecation/FutureWarning tests should be kept after change.
         self._check_inverse_of_slicing(np.array([[0, 1], [2, 1]]))
-        with warnings.catch_warnings():
-            warnings.filterwarnings('error', category=DeprecationWarning)
-            assert_raises(DeprecationWarning, delete, self.a, [100])
-            assert_raises(DeprecationWarning, delete, self.a, [-100])
+        with pytest.raises(IndexError):
+            delete(self.a, [100])
+        with pytest.raises(IndexError):
+            delete(self.a, [-100])
         with warnings.catch_warnings(record=True) as w:
             warnings.filterwarnings('always', category=FutureWarning)
             self._check_inverse_of_slicing([0, -1, 2, 2])
```

Comment:
I'm pretty sure I made a mistake in this patch, I'd expect this to produce one fewer warning, yet it appears to pass.

Reply:
Hmmm, strange, then why did this not fail? There are two `delete` calls in there, but the FutureWarning that is being detected here must not be the one that you deleted. Maybe something completely unrelated?

Reply:
Wait, below the boolean index still gives a futurewarning right? And there are two calls of `delete`. So I guess it should have checked the warnings 4 times, or reset in between the two checks simply. Seems fair enough, the boolean fixup just needs to delete it all.